src/components/Request/Pay/Views/Success.view.tsx

```diff
@@ -3,24 +3,24 @@ import * as _consts from '../Pay.consts'
 import Icon from '@/components/Global/Icon'
 import * as utils from '@/utils'
 import { useContext, useEffect, useMemo, useState } from 'react'
-import { loopUntilSuccess } from '@/components/utils/utils'
+import { fetchDestinationChain } from '@/components/utils/utils'
 import * as context from '@/context'
 
 export const SuccessView = ({ transactionHash, requestLinkData }: _consts.IPayScreenProps) => {
-    const explorerUrlWithTx = useMemo(
-        () => `${utils.getExplorerUrl(requestLinkData.chainId)}/tx/${transactionHash}`,
-        [transactionHash, requestLinkData.chainId]
+    const { selectedChainID } = useContext(context.tokenSelectorContext)
+    const sourceUrlWithTx = useMemo(
+        () => `${utils.getExplorerUrl(selectedChainID)}/tx/${transactionHash}`,
+        [transactionHash, selectedChainID]
     )
     const [explorerUrlDestChainWithTxHash, setExplorerUrlDestChainWithTxHash] = useState<
         { transactionId: string; transactionUrl: string } | undefined
     >(undefined)
     const explorerUrlAxelarWithTx = 'https://axelarscan.io/gmp/' + transactionHash
 
-    const { selectedChainID } = useContext(context.tokenSelectorContext)
 
     useEffect(() => {
         if (transactionHash) {
-            loopUntilSuccess(transactionHash, setExplorerUrlDestChainWithTxHash)
+            fetchDestinationChain(transactionHash, setExplorerUrlDestChainWithTxHash)
```

Comment:
good practice to include and will take into consideration for future effects, but in this case transactionHash won't change because we are in a success view